app/javascript/components/supplemental_application/SupplementalApplicationPage.js

```diff
@@ -211,39 +211,20 @@ class SupplementalApplicationPage extends React.Component {
     }
   }
 
-  handleSaveNewRentalAssistance = async (rentalAssistance) => {
-    // Salesforce does not accept currency strings. TODO: Move this into the form
+  handleSaveRentalAssistance = async (rentalAssistance, action = 'create') => {
+    let rentalAssistanceAction
+    // Salesforce does not accept currency strings.
     rentalAssistance['assistance_amount'] = currencyToFloat(rentalAssistance['assistance_amount'])
-    const response = await this.handleRentalAssistanceAction(apiService.createRentalAssistance)(
-      rentalAssistance,
-      this.state.persistedApplication.id
-    )
-    // show price in right format
-    rentalAssistance.assistance_amount = formUtils.formatPrice(rentalAssistance.assistance_amount)
-    if (response) {
-      rentalAssistance.id = response.id
-      this.setState(prev => {
-        return {
-          persistedApplication: {
-            ...prev.persistedApplication,
-            rental_assistances: [...prev.persistedApplication.rental_assistances, rentalAssistance]
```

Comment:
@chrisdolendo I think this is what's breaking the new rental assistance behavior

Reply:
yes great catch!